core/src/main/java/io/parsingdata/metal/Util.java

```diff
@@ -58,6 +61,11 @@ public static boolean notNullAndSameClass(final Object object, final Object othe
             && object.getClass() == other.getClass();
     }
 
+    public static BigInteger checkNotNegative(final BigInteger argument, final String name) {
+        if (checkNotNull(argument, name).compareTo(ZERO) < 0) { throw new IllegalArgumentException("Argument " + name + " may not be negative."); }
```

Comment:
..`, but was: " + argument`.

Reply:
Perhaps you should create an ```UnsignedBigInteger``` or a ```PositiveBigInteger``` class. :laughing:

Reply:
Good point but let's do this in a general review of all error messages.